### PR TITLE
[Feature] Extended coverage report

### PR DESCRIPTION
**Pull request by jpfeuffer**
In our old CMake test scripts I saw entries for building OpenMS with Coverage information.
They did not really work anymore, so I updated them and made a CMake target out of it.
Use with "caution": It uses more than 40 GB of disk space and tests are quite slow.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
wow that's big. does this include boost and stl etc.?

**Comment by jpfeuffer** [comment]
Hmm.. I am not sure. I would not think so, as I was only measuring the size of the OpenMS build tree.

**Comment by jpfeuffer** [comment]
Do you know how big a "standard" Debug build is?

File: cmake/Coverage.cmake [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# $Authors: Julianus Pfeuffer $
+# --------------------------------------------------------------------------
+
+#------------------------------------------------------------------------------
+# This cmake file enables generation of gcov files.
+
+# add compiler flag
+# -> requires clang > 3.1 or gcc > 4.8 (will be checked in module)
+if (MSVC)
+  message(WARNING "Coverage can only be enabled for GCC or Clang in a UNIX environment")
```
Comment (cbielow) [inline cmake/Coverage.cmake]: FATAL_ERROR more appropriate?

**Comment by cbielow** [comment]
+1

File: cmake/Coverage.cmake (hunk 1 above)
Comment (jpfeuffer) [inline cmake/Coverage.cmake]: Good point. Changed it.

**Comment by timosachsenberg** [comment]
+1

**Pull request merged**